form.yml

```diff
@@ -0,0 +1,326 @@
+netweaver:
+  $name: Netweaver
+  $type: group
+  
+  install_packages:
+    $name: Install required packages
+    $type: boolean
+    $default: true
+    $help: Install all required packages to install SAP Netweaver
+  
+  virtual_addresses:
+    $name: Virtual Addresses
+    $type: edit-group
+    $prototype:
+      $key:
+          $type: text
+          $name: IP Address
+      $type: text
+      $name: Hostname
+
+  sidadm_user_checkbox:
+    $name: Create the SAP sidadm user and group before the installation
+    $type: boolean
+    $default: false      
+  sidadm_user:
+    $name: SAP User/Group
+    $visibleIf: .sidadm_user_checkbox == true
+    $type: group
+    $help: If this entry exists the user and group will be created before the installation, not otherwise
+    uid:
+      $name: User ID
+      $type: text
+    gid:
+      $name: Group ID
+      $type: text
+
+  sid_adm_password_checkbox:
+    $name: Provide SAP <sid>adm user password
+    $type: boolean
+    $default: false
+    $help: <sid>adm password is optional, master password will be used as default, if the value is not defined
+  sid_adm_password:
+    $name: SAP <sid>adm user password
+    $visibleIf: .sid_adm_password_checkbox == true
+    $type: password
+    $help: The password of the SAP <sid>adm user. If not set master password will be used as default
+    $optional: true
+  
+  sap_adm_password_checkbox:
+    $name: Provide SAP <sap>adm user password
+    $type: boolean
+    $default: false
+    $help: <sap>adm password is optional, master password will be used as default, if the value is not defined
+  sap_adm_password:
+    $name: SAP admin <sap>adm user password
+    $visibleIf: .sap_adm_password_checkbox == true
+    $type: password
+    $help: The password of the SAP <sap>adm user. If not set master password will be used as default
+    $optional: true
+  
+  master_password:
```

Comment:
I would put the `master_password` before the rest of the password. I think it follows the flow better.
What do you think?

Reply:
I have followed the layout of pillar data for order of the form. Any way is fine, I can update it ask master password first